community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -0,0 +1,584 @@
+import org.neo4j.kernel.api.index.IndexDescriptor;
+import org.neo4j.kernel.api.index.InternalIndexState;
+import org.neo4j.kernel.impl.api.operations.KeyReadOperations;
+import org.neo4j.kernel.impl.core.NodeManager;
+import org.neo4j.kernel.impl.core.ThreadToStatementContextBridge;
+import org.neo4j.kernel.impl.query.QueryEngineProvider;
+import org.neo4j.kernel.impl.query.QueryExecutionEngine;
+import org.neo4j.kernel.impl.query.QueryExecutionKernelException;
+import org.neo4j.kernel.impl.store.StoreId;
+import org.neo4j.kernel.impl.traversal.BidirectionalTraversalDescriptionImpl;
+import org.neo4j.kernel.impl.traversal.MonoDirectionalTraversalDescription;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+import org.neo4j.kernel.lifecycle.LifecycleException;
+import org.neo4j.logging.Log;
+import org.neo4j.tooling.GlobalGraphOperations;
+
+import static java.lang.String.format;
+
+import static org.neo4j.collection.primitive.PrimitiveLongCollections.map;
+import static org.neo4j.helpers.collection.IteratorUtil.emptyIterator;
+import static org.neo4j.kernel.impl.api.operations.KeyReadOperations.NO_SUCH_LABEL;
+import static org.neo4j.kernel.impl.api.operations.KeyReadOperations.NO_SUCH_PROPERTY_KEY;
+
+/**
+ * Implementation of the GraphDatabaseService/GraphDatabaseService interfaces. This delegates to the
+ * services created by the {@link org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory}.
+ *
+ * To make a custom GraphDatabaseFacade, the best option is to subclass an existing GraphDatabaseFacadeFactory. Another
+ * alternative, used by legacy database implementations, is to subclass this class and call
+ * {@link org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory#newFacade(java.util.Map, org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.Dependencies, GraphDatabaseFacade)} in the constructor.
+ */
+public class GraphDatabaseFacade
+    implements GraphDatabaseAPI
+{
+    private static final long MAX_NODE_ID = IdType.NODE.getMaxValue();
+    private static final long MAX_RELATIONSHIP_ID = IdType.RELATIONSHIP.getMaxValue();
+
+    private boolean initialized = false;
+
+    private ThreadToStatementContextBridge threadToTransactionBridge;
+    private NodeManager nodeManager;
+    private IndexManager indexManager;
+    private Schema schema;
+    private AvailabilityGuard availabilityGuard;
+    private Log msgLog;
+    private LifeSupport life;
+    private Supplier<KernelAPI> kernel;
+    private Supplier<QueryExecutionEngine> queryExecutor;
+    private KernelEventHandlers kernelEventHandlers;
+    private TransactionEventHandlers transactionEventHandlers;
+
+    private long transactionStartTimeout;
+    private DependencyResolver dependencies;
+    private Supplier<StoreId> storeId;
+    protected String storeDir;
+
+    public PlatformModule platformModule;
+    public EditionModule editionModule;
+    public DataSourceModule dataSourceModule;
```

Comment:
Makes sense. I didn't notice that they were public.

We could have all access go through the module refs, although I do think there would be a (tiny) performance overhead of doing so, since it is one additional indirection.